src/main.ts

```diff
@@ -3,6 +3,7 @@
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 
 import { StarboardNotebookElement } from "./components/notebook";
+import "./initServiceWorker";
```

Comment:
This should be optional! We don't want to forcibly register a service worker for every user. I'm not sure how to best do that though.

Reply:
I'm not 100% sure either.. One could put it in the search query, but `unpkg` CDN removes any querystring which makes ad-hoc embedding more difficult (and other CDNs generally don't serve html files with the correct content type header (something like `text/html`)).

We could maybe ship two different `index.html` files, we can ship an additional one that registers the code in `initServiceWorker` prior to loading the nb at all. Still not great.. But I guess many notebooks won't need [coop/coep](http://www.foddy.net/Athletics.html)? 

We can also put it in the `runtimeConfig`, perhaps that's the cleanest approach (although it does reload rather late then.. but I guess that's a one-time pain)

Another option (sorry for the overload): we expose it under `runtime.exports.core.register<some_name>HeaderServiceWorker`, which takes argument `{autoReload: boolean}`, and returns something like `{reloadRequired: boolean}`. Then within a notebook itself you can enable it, and if you want to get fancy you can warn the user what is about to happen (and they can save their work or something).

Reply:
I decided to put it in `runtimeConfig`